VideoLocker/src/main/java/org/edx/mobile/module/db/impl/IDatabaseImpl.java

```diff
@@ -759,6 +757,30 @@ public Boolean isAnyVideoDownloadingInSubSection(final DataCallback<Boolean> cal
         return enqueue(op);
     }
 
+    @Override
+    public List<String> getUniqueCourseIdsForDownloadedVideos(final DataCallback<List<String>> callback) {
+        DbOperationGetColumn<String> op = new DbOperationGetColumn<String>(true,
+                DbStructure.Table.DOWNLOADS,
+                new String[]{DbStructure.Column.EID},
+                DbStructure.Column.USERNAME + "=? AND " + DbStructure.Column.DOWNLOADED + "=?",
+                new String[]{username(), String.valueOf(DownloadedState.DOWNLOADED.ordinal())},
+                null, String.class);
+        if (callback != null) {
+            op.setCallback(new DataCallback<List<String>>() {
+                @Override
+                public void onResult(List<String> result) {
+                    callback.sendResult(result);
+                }
```

Comment:
Why do you need a wrapper for the data callback? The executer already calls `sendResult()` instead of invoking the `onResult()` callback method directly.